core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -874,4 +874,38 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
       }
     }
   }
+
+  test("chi square test of randomizeInPlace") {
+    // Parameters
+    val arraySize = 10
+    val numTrials = 1000
+    val threshold = 0.05
+    val seed = 1L
+
+    // results[i][j]: how many times Utils.randomize moves an element from position j to position i
+    val results: Array[Array[Long]] = Array.ofDim(arraySize, arraySize)
+
+    // This must be seeded because even a fair random process will fail this test with
+    // probability equal to the value of `threshold`, which is inconvenient for a unit test.
+    val rand = new java.util.Random(seed)
+    val range = 0 until arraySize
+
+    for {
+      _ <- 0 until numTrials
+      trial = Utils.randomizeInPlace(range.toArray, rand)
```

Comment:
@srowen IMHO, @nicklavers's original `for` comprehension follows a common and well-known Scala idiom. In my mind, it's simpler and easier to understand than a nested loop.